packages/mui-base/src/SnackbarUnstyled/useSnackbar.ts

```diff
@@ -0,0 +1,199 @@
+    if (!onClose || autoHideDurationParam == null) {
+      return;
+    }
+
+    clearTimeout(timerAutoHide.current);
+    timerAutoHide.current = setTimeout(() => {
+      handleClose(null, 'timeout');
+    }, autoHideDurationParam);
+  });
+
+  React.useEffect(() => {
+    if (open) {
+      setAutoHideTimer(autoHideDuration);
+    }
+
+    return () => {
+      clearTimeout(timerAutoHide.current);
+    };
+  }, [open, autoHideDuration, setAutoHideTimer]);
+
+  const handleClickAway = (event: React.SyntheticEvent<any> | Event) => {
+    if (onClose) {
+      onClose(event, 'clickaway');
+    }
+  };
+
+  // Pause the timer when the user is interacting with the Snackbar
+  // or when the user hide the window.
+  const handlePause = () => {
+    clearTimeout(timerAutoHide.current);
+  };
+
+  // Restart the timer when the user is no longer interacting with the Snackbar
+  // or when the window is shown back.
+  const handleResume = React.useCallback(() => {
+    if (autoHideDuration != null) {
+      setAutoHideTimer(resumeHideDuration != null ? resumeHideDuration : autoHideDuration * 0.5);
+    }
+  }, [autoHideDuration, resumeHideDuration, setAutoHideTimer]);
+
+  const createHandleBlur =
+    (otherHandlers: Record<string, React.EventHandler<any> | undefined>) =>
+    (event: React.FocusEvent<HTMLDivElement, Element>) => {
+      const onBlurCallback = otherHandlers.onBlur;
+      if (onBlurCallback) {
+        onBlurCallback(event);
+      }
+      handleResume();
+    };
+
+  const createHandleFocus =
+    (otherHandlers: Record<string, React.EventHandler<any> | undefined>) =>
+    (event: React.FocusEvent<HTMLDivElement, Element>) => {
+      const onFocusCallback = otherHandlers.onFocus;
+      if (onFocusCallback) {
+        onFocusCallback(event);
+      }
+      handlePause();
+    };
+
+  const createMouseEnter =
+    (otherHandlers: Record<string, React.EventHandler<any> | undefined>) =>
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      const onMouseEnterCallback = otherHandlers.onMouseEnter;
+      if (onMouseEnterCallback) {
+        onMouseEnterCallback(event);
+      }
+      handlePause();
+    };
+
+  const createMouseLeave =
+    (otherHandlers: Record<string, React.EventHandler<any> | undefined>) =>
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      const onMouseLeaveCallback = otherHandlers.onMouseLeave;
+      if (onMouseLeaveCallback) {
+        onMouseLeaveCallback(event);
+      }
+      handleResume();
+    };
+
+  React.useEffect(() => {
+    // TODO: window global should be refactored here
+    if (!disableWindowBlurListener && open) {
+      window.addEventListener('focus', handleResume);
+      window.addEventListener('blur', handlePause);
+
+      return () => {
+        window.removeEventListener('focus', handleResume);
+        window.removeEventListener('blur', handlePause);
+      };
+    }
+
+    return undefined;
+  }, [disableWindowBlurListener, handleResume, open]);
```

Comment:
```suggestion
  }, [disableWindowBlurListener, handleResume, handlePause, open]);
```

Reply:
Looks like this was missing, no?

Reply:
No it wasn't missing. It is safe to omit `handlePause` because nothing in it references any props or state while `handleResume` does depends on some values.

- handlePause has `timerAutoHide` and we don't want to run the effect when refs change because changing refs don't re-render the component.
```js
const handlePause = () => {
  clearTimeout(timerAutoHide.current);
};
```

- handleResume depends on a few props.
```js
const handleResume = React.useCallback(() => {
    if (autoHideDuration != null) {
      setAutoHideTimer(resumeHideDuration != null ? resumeHideDuration : autoHideDuration * 0.5);
    }
  }, [autoHideDuration, resumeHideDuration, setAutoHideTimer]);
```

Also eslint also doesn't complain if we exclude it.